text/0000-bee-model-crate/0000-bee-model-crate.md

```diff
@@ -0,0 +1,334 @@
+
+            if hash.ends_with(MIN_WEIGHT_MAGNITUDE_AS_STRING){
+                transaction_metadata.hash = hash.clone();
+                break
+            }
+
+        }
+
+        (
+            Transaction {
+                signature_fragments: transaction_builder.signature_fragments.clone(),
+                address: transaction_builder.address.clone(),
+                value: transaction_builder.value.clone(),
+                obsolete_tag: transaction_builder.obsolete_tag.clone(),
+                timestamp,
+                current_index: transaction_builder.current_index.clone(),
+                last_index: transaction_builder.last_index.clone(),
+                bundle_hash: transaction_builder.bundle_hash.clone(),
+                trunk: transaction_builder.trunk.clone(),
+                branch: transaction_builder.branch.clone(),
+                nonce,
+                tag: transaction_builder.tag.clone(),
+                attachment_timestamp: transaction_builder.attachment_timestamp.clone(),
+                attachment_timestamp_lower_bound: transaction_builder.attachment_timestamp_lower_bound.clone(),
+                attachment_timestamp_upper_bound: transaction_builder.attachment_timestamp_upper_bound.clone(),
+            },
+
+            transaction_metadata
+
+        )
+
+    }
+
+}
+
+```
+
+
+### Bundle
+
+All transactions in the same bundle have the same value in the bundle field. This field contains the bundle hash, which is derived from a hash of the values of each transaction's address, value, obsoleteTag, currentIndex, lastIndex and timestamp fields.
+- **address**
+- **value**
+- **obsoleteTag**
+- **currentIndex**
+- **lastIndex**
+- **timestamp**
+
+#### Bundle Builder
+
+Similar to Transaction Builder, BundleBuilder makes it possible to create a Bundle. As mentioned, a bundle is a special collection of transactions.
+Bundles are read from head to tail but created from tail to head. This is why it makes sense to have a dedicated class for this purpose.
+The transactions inside a bundle are connected through the trunk. The trunk field of the head transaction (index 0) references the transaction with index 1, and so on.
+
+```rust
+
+struct BundleBuilder<'a> {
+
+    pub tailToHead: Vec<&'a TransactionBuilder>
+    
+}
+
+impl<'a> BundleBuilder<'a> {
+
+    pub fn append(&mut self, transaction_builder: &'a TransactionBuilder) {
+        self.tailToHead.push(transaction_builder);
+    }
+    
+    pub fn build(&self) -> Result<Bundle, BundleBuildError> {
+        
+        if self.tailToHead.size() == 0 {
+            return Err(BundleBuilder("Cannot build: bundle is empty (0 transactions)."))
+        }
+        
+        setFlags();
+        buildTrunkLinkedChainAndReturnHead();
+        
+        Ok(Bundle::new(tailToHead))
+    }
+
+}
+
+```
+
+
+# Drawbacks
+
+Without any bee-model crate, nodes can not exchange transactions. Therefore this crate seems necessary.
+
+# Rationale and alternatives
```

Comment:
An additional rationale is definitely that this kind of interface is relatively minimal and easily extended upon in a future iteration.